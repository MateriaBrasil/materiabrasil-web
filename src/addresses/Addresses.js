import React from 'react'
import Grid from '@material-ui/core/Grid'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

import New from './New'
import List from './List'

export default props => {
  const { currentUser, supplier } = props

  return (
    <Grid item xs={12} style={{ marginBottom: 36 }}>
      <Card>
        <CardContent>
          <Typography
            variant="headline"
            color="textSecondary"
            style={{ marginBottom: 16 }}
          >
            EndereÃ§os
          </Typography>
          {currentUser &&
            currentUser.id === supplier.userId && <New {...props} />}
          <List {...props} />
        </CardContent>
      </Card>
    </Grid>
  )
}
