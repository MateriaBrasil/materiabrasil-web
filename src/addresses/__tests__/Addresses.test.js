import React from 'react'
import renderer from 'react-test-renderer'

import Addresses from '../Addresses'

jest.mock('react-router-dom/Link', () => props => (
    <div {...props}>Link - {props.children}</div>
))
jest.mock('../New', () => props => (
    <div {...props}>New - {props.children}</div>
))
jest.mock('../List', () => props => (
    <div {...props}>List - {props.children}</div>
))


it('renders correctly', () => {
    const pathname = 'foo'
    const location =  { pathname }
    const props = { location }
    const tree = renderer.create(<Addresses {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
}) 

describe('with current user', () => {
    it('renders correctly', () => {
        const pathname = 'foo'
        const location =  { pathname }
        const props = { location }
        const tree = renderer.create(<Addresses currentUser={{}} supplier={{}} {...props}/>)
        expect(tree).toMatchSnapshot()
    }) 
}) 
