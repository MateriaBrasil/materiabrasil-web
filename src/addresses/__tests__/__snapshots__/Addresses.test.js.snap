// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
>
  <p
    className="MuiTypography-root-98 MuiTypography-body1-107 MuiTypography-colorInherit-127"
  >
    <div
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to={
        Object {
          "pathname": "/auth/sign-up",
          "state": Object {
            "referrer": "foo",
          },
        }
      }
    >
      Link - 
      Cadastre-se
    </div>
     
    <span>
      para ver os endere√ßos do fornecedor.
    </span>
  </p>
</div>
`;

exports[`with current user renders correctly 1`] = `
Array [
  <div
    currentUser={Object {}}
    location={
        Object {
            "pathname": "foo",
          }
    }
    supplier={Object {}}
>
    New - 
</div>,
  <div
    currentUser={Object {}}
    location={
        Object {
            "pathname": "foo",
          }
    }
    supplier={Object {}}
>
    List - 
</div>,
]
`;
